x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/stats/SearchStats.java

```diff
@@ -252,6 +254,26 @@ public boolean isIndexed(String field) {
         return stat.indexed;
     }
 
+    public boolean hasDocValues(String field) {
+        var stat = cache.computeIfAbsent(field, s -> new FieldStat());
+        if (stat.hasDocValues == null) {
+            stat.hasDocValues = false;
+            if (exists(field)) {
+                boolean hasDocValues = true;
+                for (SearchExecutionContext context : contexts) {
+                    if (context.isFieldMapped(field)) {
+                        if (context.getFieldType(field).hasDocValues() == false) {
+                            hasDocValues = false;
+                            break;
+                        }
+                    }
```

Comment:
Since exists() doesn't return the information fully, since it's the same code underneath, how about copying its implementation inside docValues and performing the hasDocValues() check as well - so there's only one SearchExecutionContext loop and the same context is called twice?

Reply:
I agree. This pattern of doing the loop for each parameter exists for a few pre-existing methods, so I kept it that way, but probably it would be better to instead capture them all in one go. I think no-one cared before because this check is not in performance critical runtime code, but in some planning code. By merging loops we would primarily share the `isFIeldMapped` and `getFieldType` calls on the `SearchExecutionContext`. All the rest if different between loops.

Reply:
OK. I separated the FieldStat into two groups, things that are simple config settings, like `exists`, `indexed`, `hasDocValues` and `hasIdenticalDelegate`, and other things that might take longer to retrieve. The first set are in a record created in a single pass loop (as you suggested). The rest are accessed on demand. I also made refactoring less flaky by not having all test versions extend the production version. I also deleted the entire unused methods for getting `min` and `max`, and think we should only add those back if (hopefully when) we use them.